infoschema/slow_log.go

```diff
@@ -77,19 +78,29 @@ func parseSlowLogFile(tz *time.Location, filePath string) ([][]types.Datum, erro
 			logutil.Logger(context.Background()).Error("close slow log file failed.", zap.String("file", filePath), zap.Error(err))
 		}
 	}()
-
-	return ParseSlowLog(tz, bufio.NewScanner(file))
+	return ParseSlowLog(tz, bufio.NewReader(file))
 }
 
 // ParseSlowLog exports for testing.
 // TODO: optimize for parse huge log-file.
-func ParseSlowLog(tz *time.Location, scanner *bufio.Scanner) ([][]types.Datum, error) {
+func ParseSlowLog(tz *time.Location, reader *bufio.Reader) ([][]types.Datum, error) {
 	var rows [][]types.Datum
 	startFlag := false
 	var st *slowQueryTuple
-	var err error
-	for scanner.Scan() {
-		line := scanner.Text()
+	finish := false
+	for {
+		if finish {
+			break
+		}
+		lineByte, err := getOneLine(reader)
+		if err == io.EOF {
+			finish = true
```

Comment:
If err == io.EOF, should we break this loop immediately ?

Reply:
done.